### DISCUSSION TITLE
Kehityksen työnjako

### DISCUSSION THREAD

**Discussion by CodoDerDritte** [KohaSuomi/Koha#1237] [Yleinen]
Tämä keskusteluketju on tarkoitettu "Kehityksen työnjako" wiki-dokumentin kommenteille: https://github.com/KohaSuomi/Koha/wiki/Kehityksen-ty%C3%B6njako

**Comment by johannaraisa** [comment]
Jos nyt tosta äkkiä laskee, niin jokaiselle tulisi n. 3 osa-aluetta. Miten tätä on tarkoitus pyörittää, jos nyt esim. ajattelee että tekisi kuvailua, hakua ja viestintää. Pyöritelläänkö peukaloita, jos ei ole mitään vai mennäänkö muiden tontille?

**Comment by lmstrand** [comment]
Ehdottomasti useamman tulisi hallita yhtä kokonaisuutta, kuin yksi devaaja (esim. Johanna jää bussin alle -> valutuksen kehitys tyssää, ainakin suht pitkäksi aikaa. Johannan äitiysloma oli hyvä esimerkki). Devaajien välistä sisäistä koulutusta? Tiketteihin reagointi pitäisi minusta silti tapahtua, kuten se nykyisellään toteutuu. Devaajat, joilla on aikaa käytettäväksi bugien/ominaisuuksien korjaukseen/kehitykseen, ottavat ne hoidettavakseen vastuutuksessa, oli devaajan osa-alue mikä tahansa. Tilanne työtehtävien ja käytettävän työajan kanssa elää niin paljon, ettei minusta voida eristää devaajia keskittymään vain omaan määrättyyn osa-alueeseensa tikettien yms. työtehtävien osalta. Tiketteihin reagointi olisi hitaampaa ja asiakkaille se näkyisi palvelun huonontumisena. Sisäistä konsultaatiota sitten eri osa-alueiden experttien suuntaan, jos tarvetta.

**Reply by lmstrand** [reply]
Kodon jako osa-alueisiin on järkevä ja looginen, eikä tuota jakoa oikein muutenkaan voisi selkeämmin tehdä.

**Comment by johannaraisa** [comment]
Minusta säännöllisten koulutusten pitäminen keskenään olisi todella tärkeää. Siinä vaiheessa varmaan dokumentointikin tulisi kuntoon, kun pitäisi kouluttaa muita. "Full stack"-devaajia tässä joudutaan kuitenkin olemaan, siitä ei päästä, mutta se että tiedetään miten asiat toimii, eikä tarvitse opetalla kaikkea alusta lähtien.

Minä mietin myös sitä, että miten vanhemman kehittäjän rooli tässä sopii tuon kanssa yhteen. Pitäisikö meillä olla isommat alueet ja ohjataan niissä sitä kehitystä?

**Reply by CodoDerDritte** [reply]
> Minä mietin myös sitä, että miten vanhemman kehittäjän rooli tässä sopii tuon kanssa yhteen. Pitäisikö meillä olla isommat alueet ja ohjataan niissä sitä kehitystä?

Ajattelin että se ois tuota "Järjestelmäarkkitehtuuri ja kehityksen linjaaminen järjestelmänäkökulmasta" hommaa, mutta siihen taitaa liittyä muutakin, tarvii tuumata.

**Comment by emta001** [comment]
Lisäilin tuonne tilastoinnin osalta muutaman osa-alueen. Laitoin sinne myös raporttien lisäämisen/päivittämisen. Niistä taisi olla sovittu niin, että pääkäyttäjät niitä tekevät omiin/kimpan tarpeisiin ja laittavat mahdollisesti muillekin jakoon (korjatkaa jos olen väärässä). Mutta toisinaan niiden kanssa voidaan tarvita kehittäjien apua.

**Comment by johannaraisa** [comment]
Olen tässä pohtinut tätä ja ei näihin ihan selkeää työnjakoa pysty tekemään, tehdessä joutuu hoitamaan useampaa osa-aluetta. Kuten tikettiä tehtäessä se olisi hyvä saada samalla yhteisön tietoisuuteen/versioon ja ja loogisinta on, että sen hoitaa sama henkilö joka on tehnyt tiketin.

Näkisin sen hyödyllisemmäksi, että viikon työlistalle määritellään joku joukko hommia ja sitten seurataan niiden etenemistä. Yleensähän noita tikettejä voi olla työnalla useampi, jos joku ei edisty, niin pystyy tekemään toista.
